capsules/extra/src/process_info_driver.rs

```diff
@@ -0,0 +1,254 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2025.
+
+//! Allow userspace to inspect and control processes on the board.
+//!
+//! ## Warning!
+//!
+//! This capsule is designed for testing and experimental use cases only. It
+//! should not be used in production! It was originally written for use in an
+//! educational tutorial to make it easy to interact with processes stored on
+//! the board from userspace using a screen.
+//!
+//! This capsule does require a capability to also indicate that this interacts
+//! with processes in a way that common capsules should not.
+//!
+//! ## Commands
+//!
+//! - 0: Check driver exists.
+//! - 1: Get the count of processes running on the board.
+//! - 2: Fill the allow RW buffer with the process IDs for the running processes
+//!   on the board. Returns the number of running processes.
+//! - 3: Fill the allow RW buffer with the short IDs for the running processes
+//!   on the board. Returns the number of running processes.
+//! - 4: Put the name of the process specified by the process ID in `data1` in
+//!   the allow RW buffer.
+//! - 5: Fill the allow RW buffer with the following information for the process
+//!   specified by the process ID in `data1`.
+//!   - The number of timeslice expirations.
+//!   - The number of syscalls called.
+//!   - The number of restarts.
+//!   - The current process state (running=0, yielded=1, yieldedfor=2,
+//!     stopped=3, faulted=4, terminated=5).
+//! - 10: Change the process state. `data1` is the process ID, and `data2 is the
+//!   new state.(1=start, 2=stop, 3=fault, 4=terminate, 5=boot).
```

Comment:
Why the offset from the previous command number?

Yes, this is the first command to actually manipulate process state, but I think it's a bad idea for us to only reserve 4 additional IDs (between 5 and 10) for inspecting process state. Maybe assigning consecutive command numbers is fine?

Reply:
Changed to command num 6.